derive/src/utils.rs

```diff
@@ -179,22 +189,40 @@ pub fn codec_crate_path(attrs: &[Attribute]) -> syn::Result<Path> {
 	}
 }
 
-/// Trait bounds.
-pub type TraitBounds = Punctuated<syn::WherePredicate, token::Comma>;
-
-/// Parse `name(T: Bound, N: Bound)` as a custom trait bound.
-struct CustomTraitBound<N> {
-	_name: N,
-	_paren_token: token::Paren,
-	bounds: TraitBounds,
+/// Parse `name(T: Bound, N: Bound)` or `name(skip_type_params(T, N))` as a custom trait bound.
+pub enum CustomTraitBound<N> {
+	SpecifiedBounds {
+		_name: N,
+		_paren_token: token::Paren,
+		bounds: Punctuated<syn::WherePredicate, token::Comma>,
```

Comment:
This I think should stay as `WherePredicate`, because that's exactly what we're substituting, and technically speaking there are things that a simple `TypeParam` can't do, such as the HRTB `for<'a>` syntax.

Reply:
Good point.